### PR TITLE
Setup test with extension file

### PR DESCRIPTION
**Pull request by vickcoo**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description
This PR will set up unit tests, and create an initial unit test for `extension.swift`, This will ensure that everything is working correctly for the tests, and then we can proceed to create additional tests.

### Related issue

- #8

### CODE REVIEW

**Comment by vickcoo** [comment]
Hi @andrewtavis , this PR is a work in progress, I'll notify you when it's finished.

@fabulouiOS-monk,  I've started working on the unit tests and need your help.

**Comment by fabulouiOS-monk** [comment]
Sure @vickcoo, I will be in touch with you! 😄

**Comment by fabulouiOS-monk** [comment]
@vickcoo, I think some unwanted **Scribe.xcodeproj** changes have also been pushed.

Whenever you push changes regarding the above file it should only contain any new file modification (either deletion or addition of a file). In future to avoid this you can make use of VScode to revert the changes from **Scribe.xcodeproj** since from the terminal it is impossible to do so.

@andrewtavis, Please suggest if those changes are required or @vickcoo can revert those changes.

**Comment by andrewtavis** [comment]
Are we planning on creating a GitHub action to run these on PRs? I think that this would add a lot to this process :)

**Comment by andrewtavis** [comment]
For `Scribe.xcodeproj/project.pbxproj` the addition of the testing file should change the this file, but I'm a bit confused where all the `xcschemas` are coming from 🤔

**Comment by vickcoo** [comment]
@fabulouiOS-monk thanks, I'll be careful with the `project.pbxproj` file. 😊

**Comment by vickcoo** [comment]
@andrewtavis those files were automatically created when I create a test target

**Comment by fabulouiOS-monk** [comment]
> Are we planning on creating a GitHub action to run these on PRs? I think that this would add a lot to this process :)

Are you referring to the **xcschemas**, ]No we don't need to devs can just exclude those from the commit.

**Comment by andrewtavis** [comment]
Do we want to remove them from the PR and then put a * reference to ignore files like them in `.gitignore`?

**Comment by fabulouiOS-monk** [comment]
> Do we want to remove them from the PR and then put a * reference to ignore files like them in `.gitignore`?

Yup, I think it is good to do that. Unless we really need some changes on those files.

**Comment by vickcoo** [comment]
I've reverted the `xcscheme`, which seems to work well for tests.

**Comment by andrewtavis** [comment]
Want to note again that it would be great if we figure out some kind of CI run for this where PRs will run the tests and we all can then see if the changes are breaking anything :)

**Comment by andrewtavis** [comment]
We can also do a new issue for this, if need be! What are your two's thoughts on this?

**Comment by fabulouiOS-monk** [comment]
> Want to note again that it would be great if we figure out some kind of CI run for this where PRs will run the tests and we all can then see if the changes are breaking anything :)

Yup, lets find out. You can create an issue and ping me.

File: KeyboardsTests/KeyboardsBase/ExtensionsTest.swift [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    XCTAssertEqual(result, expectedResult)
+  }
+}
+
+// MARK: deletePriorToCursor
+
+extension ExtensionTest {
+  func testDeletePriorToCursor() {
+    let string = "Hello"
+    let expectedResult = "Hel│"
```
Comment (andrewtavis) [inline KeyboardsTests/KeyboardsBase/ExtensionsTest.swift:110]: Note that this should be `Hell│` as we're just deleting once, right? (I'm fine with this being the result, btw) 😇

**Comment by andrewtavis** [comment]
One thing that I'm noticing here is that we're mixing pascal case and snake case for the names of the test functions? Is this norma, or can I rename them all with full pascal case? I'd fix the comment above as well :)

**Comment by andrewtavis** [comment]
I.e. I'd change `testUnique_uniqueElements_withoutDuplicates` to `testUniqueElementsWithoutDuplicates` and the like :)

Thoughts on this, @vickcoo and @fabulouiOS-monk :)

**Comment by fabulouiOS-monk** [comment]
> One thing that I'm noticing here is that we're mixing pascal case and snake case for the names of the test functions? Is this norma, or can I rename them all with full pascal case? I'd fix the comment above as well :)

I suggested to Vickcoo to name it like this so that it's a little bit more readable than what this test case is expected to produce.

**Comment by fabulouiOS-monk** [comment]
If 'testUniqueElementsWithoutDuplicates' makes it good. What I thought is for other devs to understand easily just from one look at the test case name: "functionName_WhatWeExpect", So later on when some other devs are searching for a particular test case, it will be easier for them.

**Comment by andrewtavis** [comment]
Edit, above I mean `camelCase`, not `PascalCase`.

I appreciate the idea here, @fabulouiOS-monk! Logically it makes sense, but the syntax is something we'll need to explain to people consistently and might get confusing. If it's not a Swift convention, then I hope it's ok that I switch them to full camel case. It was kind of difficult for me to read them at first glance, and we should be following the coding language's standards here :)

**Comment by fabulouiOS-monk** [comment]
@andrewtavis, Yes you are correct, we have to follow coding language standards 😄.

File: KeyboardsTests/KeyboardsBase/ExtensionsTest.swift (hunk 1 above)
Comment (vickcoo) [reply KeyboardsTests/KeyboardsBase/ExtensionsTest.swift:110]: Yes, that's deleting it once!

**Comment by vickcoo** [comment]
> I appreciate the idea here, @fabulouiOS-monk! Logically it makes sense, but the syntax is something we'll need to explain to people consistently and might get confusing. If it's not a Swift convention, then I hope it's ok that I switch them to full camel case. It was kind of difficult for me to read them at first glance, and we should be following the coding language's standards here :)

No problem, I'll modify the name to adhere swift conventions. 🚀 
Thanks @andrewtavis, that's a good idea and it make sense. I also appreciate @fabulouiOS-monk ideas, that's  a good thought 😄

**Comment by andrewtavis** [comment]
I have the changes locally, @vickcoo :) Will send them along shortly!

**Review by andrewtavis — APPROVED** [review]
Ok, so I'm hoping we're good here 😅 I made the directory just `Tests`, and I think we can then mirror the project directory underneath like `Tests/Keyboards/KeyboardsBase` and `Tests/Scribe/...`.

Renamed the functions and did some minor edits to the arguments they're using. Hope the changes are ok!

Thank you both for all the hard work here and the great teamwork! Is annoying now, but the work we do here means that the app has a sustainable future. Really appreciate you two being willing to do the intangible work that's needed 💙

**Pull request merged**